apps/todos/src/app/features/todos/todos.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnInit, OnDestroy, TemplateRef } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
+import { ApiService } from '../../core/services';
+import { Todo, AddTodo } from '@learning-workspace/api-interfaces';
+
+@Component({
+  selector: 'learning-workspace-todos',
+  templateUrl: './todos.component.html',
+  styleUrls: ['./todos.component.scss'],
+})
+export class TodosComponent implements OnInit, OnDestroy {
+  public todos: Todo[] = [];
+  public selectedTodo: Todo;
+  public openAddEditForm = false;
+  public editMode = false;
+
+  private subscription = new Subscription();
+
+  constructor(private readonly apiService: ApiService) {}
+
+  ngOnInit(): void {
+    this.getTodosSubscription();
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  public addTodo(): void {
+    this.editMode = false;
+    this.openAddEditTodoForm();
+  }
+
+  public removeTodo(id: string): void {
+    this.removeTodoSubscription(id);
+  }
+
+  public editTodo(id: string): void {
+    this.editMode = true;
+    this.selectedTodo = this.todos.find((todo) => todo.id === id) as Todo;
+    this.openAddEditTodoForm();
+  }
+
+  public checkTodo(id: string): void {
+    this.checkTodoSubscription(id);
+  }
+
+  public openAddEditTodoForm() {
```

Comment:
please add return function type to all functions in this component

Reply:
ready